src/python/pants/engine/target.py

```diff
@@ -808,14 +809,81 @@ class AllTargetsRequest:
 # Target generation
 # -----------------------------------------------------------------------------------------------
 
-_Tgt = TypeVar("_Tgt", bound=Target)
+
+class TargetGenerator(Target):
+    """A Target type which generates other Targets via installed `@rule` logic.
+
+    To act as a generator, a Target type should subclass this mixin and install generation `@rule`s
+    which consume a corresponding GenerateTargetsRequest subclass to produce GeneratedTargets.
+    """
+
+    # The generated Target class.
+    generated_target_cls: ClassVar[type[Target]]
+
+    # Fields which have their values copied from the generator Target to the generated Target.
+    #
+    # Must be a subset of `core_fields`.
+    copied_fields: ClassVar[Tuple[Type[Field], ...]]
+
+    # Fields which are specified to instances of the generator Target, but which are propagated
+    # to generated Targets rather than being stored on the generator Target.
+    #
+    # Must be disjoint from `core_fields`.
+    moved_fields: ClassVar[Tuple[Type[Field], ...]]
```

Comment:
I'd appreciate a few more words on the concept of copied vs moved fields. What are the implications, when do I want to include a field in moved, or copied or neither?

Reply:
+1, guidance on when to use what.

Reply:
Yea, good idea.